### ISSUE TITLE
CS1585	Member modifier 'replace' must precede the member type and name

### ISSUE THREAD

**Issue by basoundr** [dotnet/roslyn#11503]
**Version Used**:

Bug seen in Roslyn Future  20160520.2 

**Steps to Reproduce**:
Copy and paste the below code

``` C#
using System.Text.RegularExpressions;

namespace TestCS1585
{
    public class Repro
    {
        public static string replace(string input, string regexp, string flags, string replacement)
        {

            RegexOptions options = RegexOptions.ECMAScript;

            if (flags.IndexOf("i") != -1)
            {
                options |= RegexOptions.IgnoreCase;
            }

            Regex regex = new Regex(regexp, options);

            if (flags.IndexOf("g") != -1)
            {
                return regex.Replace(input, replacement, 1);
            }
            else
            {
                return regex.Replace(input, replacement);
            }
        }
    }
}
```

**Expected Behavior**:

No errors

**Actual Behavior**:

Repro.cs(12,30,12,37): error CS1585: Member modifier 'replace' must precede the member type and name
Repro.cs(12,30,12,37): error CS1520: Method must have a return type

**Comment by jaredpar** [comment]
@VSadov please atke a look at this.

**Comment by cston** [comment]
Fixed in #11519.
